File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.js
/**
 * @param {string} s
 * @return {string}
 */
var removeOuterParentheses = function(s) {
    let result = "";
        let opened = 0;
        
        for (const char of s) {
            if (char === '(') {
                opened++;
                if (opened > 1) {
                    result += char;
                }
            } else {
                opened--;
                if (opened > 0) {
                    result += char;
                }
            }
        }
        
        return result;
};

File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        string fin = "";
        int opened = 0;
        for(auto i : s){
            if (i == '('){
                opened++;
                if (opened > 1){
                    fin.push_back(i);
                }
            }
            else{
                opened--;
                if (opened > 0){
                    fin.push_back(i);
                }
            }
        }
        return fin;
    }
    
};